subprojects/docs/src/docs/userguide/organizingGradleProjects.adoc

```diff
@@ -15,55 +15,122 @@
 [[organizing_gradle_projects]]
 == Organizing Gradle Projects
 
-The build logic and source code of every project should be organized in a meaningful way to improve readability and avoid common problems. This page lays out the best practices by example.
+Source code and build logic of every software project should be organized in a meaningful way. This page lays out the best practices that lead to readable, maintainable projects. The following sections also touch on common problems and how to avoid them.
 
-[[sec:configuring_using_external_script]]
-=== Configuring the project using an external build script
+[[sec:separate_language_source_files]]
+=== Separate language-specific source files
 
-You can configure the current project using an external build script. All of the Gradle build language is available in the external script. You can even apply other scripts from the external script.
+Gradle's language plugins establish conventions for discovering and compiling source code. For example, a project applying the <<java_plugin,Java plugin>> will automatically compile the code in the directory `src/main/java`. Other language plugins follow the same pattern. The last portion of the directory path usually indicates the expected language of the source files.
 
-Build scripts can be local files or remotely accessible files downloaded via a URL.
+Some compilers are capable of cross-compiling multiple languages in the same source directory. The Groovy compiler can handle the scenario of mixing Java and Groovy source files located in `src/main/groovy`. It is recommended practice to create source files in the conventional directories if you work on a project that with multiple languages. Separated directories for dedicated language source files help with maintaining complex projects.
```

Comment:
"It is recommended practice to ... complex projects" 

I think it'd be better to say that "Gradle recommends that you place sources in directories according to their language, because builds are more performant and both the user and build can make stronger assumptions"

Reply:
done